elyra/pipeline/processor_airflow.py

```diff
@@ -0,0 +1,254 @@
+                cos_client.upload_file_to_dir(dir=cos_directory,
+                                              file_name=operation_artifact_archive,
+                                              file_path=dependency_archive_path)
+            except BaseException:
+                self.log.error("Error uploading artifacts to object storage.", exc_info=True)
+                raise
+
+            self.log.info("Pipeline dependencies have been uploaded to object store")
+
+        ordered_notebook_ops = OrderedDict()
+
+        while notebook_ops:
+            for i in range(len(notebook_ops)):
+                notebook = notebook_ops.pop(0)
+                if not notebook['dependencies']:
+                    ordered_notebook_ops[notebook['id']] = notebook
+                    self.log.debug("This is a root node added : %s", ordered_notebook_ops[notebook['id']])
+                elif all(deps in ordered_notebook_ops.keys() for deps in notebook['dependencies']):
+                    ordered_notebook_ops[notebook['id']] = notebook
+                    self.log.debug("This is a dependent node added : %s", ordered_notebook_ops[notebook['id']])
+                else:
+                    notebook_ops.append(notebook)
+
+        return ordered_notebook_ops
+
+    def create_pipeline_file(self, pipeline, pipeline_export_format, pipeline_export_path, pipeline_name):
+
+        runtime_configuration = self._get_runtime_configuration(pipeline.runtime_config)
+        api_endpoint = runtime_configuration.metadata['api_endpoint']
+
+        self.log.info('Creating pipeline definition as a .' + pipeline_export_format + ' file')
+        if pipeline_export_format == "json":
+            with open(pipeline_export_path, 'w', encoding='utf-8') as file:
+                json.dump(pipeline_export_path, file, ensure_ascii=False, indent=4)
+        else:
+            # Load template from installed elyra package
+            loader = PackageLoader('elyra', 'templates/airflow')
+            template_env = Environment(loader=loader)
+
+            template = template_env.get_template('airflow_template.jinja2')
+
+            notebook_ops = self._cc_pipeline(pipeline, pipeline_name)
+
+            python_output = template.render(operations_list=notebook_ops,
+                                            pipeline_name=pipeline_name,
+                                            namespace='default',
+                                            kube_config_path=None,
+                                            in_cluster='True',
+                                            api_endpoint=api_endpoint,
+                                            pipeline_description="Elyra Pipeline")
+
+            # Write to python file and fix formatting
+            with open(pipeline_export_path, "w") as fh:
+                fh.write(autopep8.fix_code(python_output))
+
+        return pipeline_export_path
+
+    def _artifact_list_to_str(self, pipeline_array):
```

Comment:
All of the following methods appear to be identical to those in [`KfpPipelineProcessor`](https://github.com/elyra-ai/elyra/blob/master/elyra/pipeline/processor_kfp.py#L220-L259) and should be moved to `PipelineProcessor`.